packages/broker/src/plugins/storage/StorageConfig.ts

```diff
@@ -43,22 +43,26 @@ export class StorageConfig {
 
     streamKeys: Set<StreamKey>
     listeners: StorageConfigListener[]
-    nodeId: string
+    clusterId: string
+    clusterSize: number
+    myIndexInCluster: number
     apiUrl: string
     private _poller!: ReturnType<typeof setTimeout>
     private _stopPoller: boolean
 
     // use createInstance method instead: it fetches the up-to-date config from API
-    constructor(nodeId: string, apiUrl: string) {
+    constructor(clusterId: string, clusterSize: number, myIndexInCluster: number, apiUrl: string) {
```

Comment:
> What happens if the clusterSize config differs between nodes? e.g. config updated but nodes not restarted at the same time

In that case the nodes aren't aligned on who handles what. Some streams may be subscribed to by multiple nodes (not a fatal problem, see below), while some streams may not be subscribed to by any nodes (the bigger problem). It's the responsibility of the storage cluster operator to ensure that this does not happen (which is acceptable for now IMO).

> What happens if there are multiple nodes with the same myIndexInCluster?

In that case the nodes will subscribe to the exact same subset of streams, and all that data gets written to Cassandra multiple times (first written and then overwritten). It will work, but produces unnecessary load and cleanup overhead for Cassandra.